pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```
@@ -1539,11 +1569,57 @@ public void setClob(@Positive int parameterIndex,
     throw Driver.notImplemented(this.getClass(), "setClob(int, Reader)");
   }
 
+  private void setBlobBytea(@Positive int parameterIndex,
+      @Nullable InputStream inputStream, @NonNegative long length)
+      throws SQLException {
+
+    if (inputStream == null) {
+      setNull(parameterIndex, Types.VARBINARY);
+      return;
+    }
+
+    byte [] b;
+
+    try {
+      // for JDK 9+ we could use b = inputStream.readNBytes(length). However ...
+      ByteArrayOutputStream os = new ByteArrayOutputStream();
+      byte[] buffer = new byte[4096];
+      long remaining = length;
+      int toRead = remaining > buffer.length ? buffer.length : (int) remaining;
+      int len = inputStream.read(buffer, 0, toRead);
+      while (remaining > 0 && len > -1) {
+        remaining -= len;
+        os.write(buffer, 0, len);
+        toRead = remaining > buffer.length ? buffer.length : (int) remaining;
+        len = inputStream.read(buffer, 0, toRead);
+      }
+      if (remaining > 0) {
+        // not enough bytes
+      }
+      b = os.toByteArray();
+    } catch (IOException ioe) {
+      throw new PSQLException(GT.tr("Provided InputStream failed."), PSQLState.UNEXPECTED_ERROR,
+          ioe);
+    }
+
+    if (b.length > PgBlobBytea.MAX_BYTES) {
+      // XXX throw exception
```

We should throw the exception here...

---

will fix